src/extension.ts

```diff
@@ -31,55 +30,45 @@ export function activate(context: vscode.ExtensionContext) {
     context.subscriptions.push(vscode.commands.registerCommand('okteto.create', createCmd));
 }
 
-function installCmd(upgrade: boolean): Promise<string> {
+function installCmd(upgrade: boolean): Promise<void> {
     let title = "Installing Okteto";
     if (upgrade) {
         title = "Okteto is out of date, upgrading";
     }
 
-    return new Promise<string>((resolve, reject) => {
+    return new Promise<void>((resolve, reject) => {
         vscode.window.withProgress({
         location: vscode.ProgressLocation.Notification,
         title: title,
     }, (progress, token)=>{
         reporter.track(events.install);
-        const p = install();
+        const p = okteto.install();
         p.then(() => {
             //vscode.window.showInformationMessage(`Okteto was successfully installed`);
             resolve();
-        }, (reason) => {
+        }, (err) => {
             reporter.track(events.oktetoInstallFailed);
-            vscode.window.showErrorMessage(`Okteto was not installed: ${reason.message}`);
-            reject();
+            reporter.captureError(`okteto was not installed: ${err}`, err);
+            vscode.window.showErrorMessage(`Okteto was not installed: ${err.message}`);
+            reject(err);
         }
         );
         return p;
     });
   });
 }
 
-function install(): Promise<string>{
-    return new Promise((resolve, reject) => {
-        okteto.install()
-        .then(()=>{
-            console.log("okteto was successfully installed");
-            resolve();
-        }, (reason) => {
-            console.error(`okteto was not installed: ${reason}`);
-            throw new Error(reason);
-        }).catch((reason) => {
-            console.error(`okteto was not installed: ${reason}`);
-            reject();
-        });
-    });
-}
-
 async function downCommand() {
-    reporter.track(events.down);
     const { install, upgrade } = okteto.needsInstall();
     if (install){
-        await installCmd(upgrade);
+        try {
+            await installCmd(upgrade);
+        } catch {
+            // error already handled on installCmd
```

Comment:
Then why do you need the try / catch?

Reply:
I put it to handle the case where `installCmd` rejects the promise. I don't want this to bubble up as a unhandled exception to vscode.

Reply:
also, if the install fails I want to stop the execution of the command.